### DISCUSSION TITLE
Documentation on how the turbo drive adapter works

### DISCUSSION THREAD

**Discussion by ajaishankar** [SimoTod/alpine-turbo-drive-adapter#60] [Q&A]
Hi @SimoTod 

A small explanation of how the adapter works would be very helpful...

Did go through the bridge code, but not able to understand the internals.

Thanks!

Ajai

**Comment by SimoTod** [comment]
The adapter adds custom html attributes to do stuff like removing html elements generated by Alpine (if and for). Turbo clones the dom element which is an operation that removes custom alpine properties so for example, after restoring from the turbo cache, if you don't clean up, you would get duplicate elements becausa Alpine doesn't know about the ones coming from the Turbo cache

**Reply by ajaishankar** [reply]
Got it, thanks!

One more question:

Does the adapter also handle turbo-frame/turbo-stream updates - or is that automatically taken care of by Alpine?
